src/scripts/librarymenu.js

```diff
@@ -155,297 +302,669 @@ define(["dom", "layoutManager", "inputManager", "connectionManager", "events", "
-            isEditorPage = page.classList.contains("metadataEditorPage"),
-            isMySyncPage = page.classList.contains("mySyncPage"),
-            id = isLiveTvPage || isChannelsPage || isEditorPage || isMySyncPage || page.classList.contains("allLibraryPage") ? "" : getTopParentId() || "",
-            elems = document.getElementsByClassName("lnkMediaFolder");
+        var i;
+        var length;
+        var isLiveTvPage = page.classList.contains("liveTvPage");
+        var isChannelsPage = page.classList.contains("channelsPage");
+        var isEditorPage = page.classList.contains("metadataEditorPage");
+        var isMySyncPage = page.classList.contains("mySyncPage");
+        var id = isLiveTvPage || isChannelsPage || isEditorPage || isMySyncPage || page.classList.contains("allLibraryPage") ? "" : getTopParentId() || "";
+        var elems = document.getElementsByClassName("lnkMediaFolder");
+
         for (i = 0, length = elems.length; i < length; i++) {
-            var lnkMediaFolder = elems[i],
-                itemId = lnkMediaFolder.getAttribute("data-itemid");
-            isChannelsPage && "channels" == itemId ? lnkMediaFolder.classList.add("navMenuOption-selected") : isLiveTvPage && "livetv" == itemId ? lnkMediaFolder.classList.add("navMenuOption-selected") : isEditorPage && "editor" == itemId ? lnkMediaFolder.classList.add("navMenuOption-selected") : isMySyncPage && "manageoffline" == itemId && -1 != window.location.href.toString().indexOf("mode=download") ? lnkMediaFolder.classList.add("navMenuOption-selected") : isMySyncPage && "syncotherdevices" == itemId && -1 == window.location.href.toString().indexOf("mode=download") ? lnkMediaFolder.classList.add("navMenuOption-selected") : id && itemId == id ? lnkMediaFolder.classList.add("navMenuOption-selected") : lnkMediaFolder.classList.remove("navMenuOption-selected")
+            var lnkMediaFolder = elems[i];
+            var itemId = lnkMediaFolder.getAttribute("data-itemid");
+
+            if (isChannelsPage && "channels" == itemId) {
+                lnkMediaFolder.classList.add("navMenuOption-selected");
+            }
+            else if (isLiveTvPage && "livetv" == itemId) {
+                lnkMediaFolder.classList.add("navMenuOption-selected");
+            }
+            else if (isEditorPage && "editor" == itemId) {
+                lnkMediaFolder.classList.add("navMenuOption-selected");
+            }
+            else if (isMySyncPage && "manageoffline" == itemId && -1 != window.location.href.toString().indexOf("mode=download")) {
+                lnkMediaFolder.classList.add("navMenuOption-selected");
+            }
+            else if (isMySyncPage && "syncotherdevices" == itemId && -1 == window.location.href.toString().indexOf("mode=download")) {
+                lnkMediaFolder.classList.add("navMenuOption-selected");
+            }
+            else if (id && itemId == id) {
+                lnkMediaFolder.classList.add("navMenuOption-selected");
```

Comment:
all above `if`-s effect in same class added to `lnkMediaFolder`, maybe combine them all in one `if`?

Reply:
They used to be a huge amount of nested if/elses. Unsure of how its actually executed. If you or someone else wants to suggest a change, I'll accept it.

Just did the lazy thing and undid 10+ layers of nesting :)